src/Catalyst.Core.Lib/Validators/TransactionValidator.cs

```diff
@@ -60,12 +60,13 @@ private bool ValidateTransactionSignature(TransactionBroadcast transactionBroadc
                 transactionBroadcast.PublicEntries.First().Base.SenderPublicKey.ToByteArray());
 
             var signingContext = transactionBroadcast.Signature.SigningContext.ToByteArray();
-            
+
             // we need to verify the signature matches the message, but transactionBroadcast contains the signature and original data,
             // passing message+sig will mean your verifying an incorrect message and always return false, so just null the sig.
-            transactionBroadcast.Signature = null;
-            
-            if (_cryptoContext.Verify(transactionSignature, transactionBroadcast.ToByteArray(), signingContext))
+            var transactionBroadcastClone = transactionBroadcast.Clone();
```

Comment:
The function only returns a bool not the modified object. Can you point out where your seeing the sig being nulled after this is being called, as it would be good to find how it was missed in the tests

Reply:
TransactionReceivedEvent.cs OnTransactionReceived after var transactionValid = _validator.ValidateTransaction(transaction); transaction.Signature will be null.